crates/oxc_cli/src/git.rs

```diff
@@ -0,0 +1,19 @@
+use std::path::Path;
+
+pub struct Git {
+    repo: git2::Repository,
+}
+
+impl Git {
+    pub fn new<P: AsRef<Path>>(path: P) -> Self {
+        let repo = git2::Repository::discover(path).unwrap();
+        Self { repo }
+    }
+
+    pub fn is_uncommitted(&self, path: &Path) -> bool {
+        match self.repo.status_file(path) {
+            Ok(status) => matches!(status, git2::Status::WT_NEW),
+            _ => false,
```

Comment:
Maybe this should be `true` since it indicates the file could not be found in HEAD/index/working tree, in which case we should filter it out from linting?